crates/net/network/src/manager.rs

```diff
@@ -112,6 +112,7 @@ pub struct NetworkManager<C> {
     metrics: NetworkMetrics,
     /// Disconnect metrics for the Network
     disconnect_metrics: DisconnectMetrics,
+    _phantom: std::marker::PhantomData<C>,
```

Comment:
we should be able to get rid of this as well, if not possible due to some downstream requirements, I'm okay with keeping it in this pr

Reply:
can we not move the generic to the function itself:

```
 pub async fn new<C>(config: NetworkConfig<C>)

```